playbooks/configure-apt-sources.yml

```diff
@@ -151,16 +185,12 @@
         - apt_artifact_found | bool
         - apt_artifact_enabled | bool
 
-    - name: Remove extra sources
-      lineinfile:
-        dest: /etc/apt/sources.list
-        state: absent
-        regexp: "{{ item }}"
-      with_items:
-        - "^deb-src"
-        - "-backports"
-        - "-security"
-        - "-updates"
+    - name: Set the preferred repo to use to be the apt artifact repo
+      copy:
+        content: >-
+          APT::Default-Release "{{ rpc_release }}-{{ ansible_distribution_release }}";
```

Comment:
What is the reason for using apt config instead of an actual apt pin? Doesn't https://github.com/rcbops/rpc-openstack/pull/2720 accomplish much of the same thing and doesn't apt-package-pin role with a high priority do what we need?

Reply:
It sets the priority of the apt artifact repo to 990, which is above the default of 500 for the other repositories. It's effectively the same as the pin, but this is the recommended way in Ubuntu/Debian docs to define one repo as the explicitly preferred source.

Reply:
As it turns out, no form of pinning I could work out resulted in a working deployment when the RPC-O repository has older packages than the updates/security repository. I've tried pinning at various levels - 501 or more. In every case this results in a broken build because it want to use the older packages. The solution therefore has to be to have the RPC-O repo at the same priority as the rest of the repositories on the system.

This will give unpredictable results, but it helps day one and does assist a transition process which can be done later to 'strict' mode.